### ISSUE TITLE
Basic example code - improve the code dealing with tasks

### ISSUE THREAD

**Issue by tobixen** [python-caldav/caldav#269]
Ref #256, the example code covering tasks is a bit hidden and may be improved.  See https://github.com/python-caldav/caldav/issues/256#issuecomment-1416494723 for details

**Comment by vrii14** [comment]
Hi, 
I would like to work on this issue. 
Just wanted to check my understanding of what has to be done - 
Examples have to be added in the basic_usage_examples for new argument in complete method, uncomplete method, set_duration, get_due and set_due method.
Is that correct?

**Comment by tobixen** [comment]
Yes.  There is already a little bit of code at the bottom of `search_calendar_demo`, it should be moved out to a separate function (`tasks_demo` or something like that).

As for `set_duration`, `get_due` and `set_due`, I'm a bit hesitant if it should be in the examples, I'm not much happy with the API so it may change in version 2.0.  At least it's needed with a warning.  (the logic should be moved to the icalendar library, ref https://github.com/collective/icalendar/issues/496 - and the logic should also apply to events, not only tasks - but since events have DTEND instead of DUE, the current code cannot simply be moved up to the base class)

**Comment by vrii14** [comment]
Okay I will work on this issue and then create a PR

**Comment by tobixen** [comment]
Perhaps a separate example file, or tutorial/howto on "advanced task handling" is appropriate.  Anyway, I have made some upgrades both to the examples and documentation, and as for now I feel there isn't much more to add when it comes to tasks.  Advanced task handling is a bit outside the scope of the caldav library.  I've added some functionality in the caldav library to support the work on my `plann` utility, but I'm not sure if the changes I've done are good or appropriate to keep in the caldav library (perhaps I will create a separate library for "advanced calendaring operations").

So everything considered, I think there is no more to do on this issue, so I will close it.
